system/include/emscripten/bind.h

```diff
@@ -1919,14 +1919,8 @@ class class_ {
 #if __cplusplus >= 201703L
 template<typename T>
 void register_optional() {
-    // Optional types are automatically registered for some internal types so
-    // only run the register method once so we don't conflict with a user's
-    // bindings if they also register the optional type.
-    thread_local bool hasRun;
-    if (hasRun) {
-        return;
-    }
-    hasRun = true;
+    // Optional types are automatically registered for some internal types; the
```

Comment:
Allowing duplicate registrations will lead to a slightly strange behavior where the order matters. e.g.

The following will fail:
```c++
  register_optional<Foo*>();
  class_<std::optional<Foo*>>("OptionalFoo");
  ```
but swapping them will work:
```c++
  class_<std::optional<Foo*>>("OptionalFoo");
  register_optional<Foo*>();
  ```

I guess if a user is creating their own binding for std::optional and they want to override the behavior that could be fine.